vespalib/src/tests/component/component.cpp

```diff
@@ -197,6 +198,79 @@ TEST("testSimple")
     EXPECT_TRUE(vs.matches(v));
 }
 
+TEST("testZeros")
+{
+    // test Version:
+
+    Version v000(0, 0, 0, "");
+    EXPECT_EQUAL("", v000.toAbbreviatedString());
+    EXPECT_EQUAL("0.0.0", v000.toString());
+    Version v000s("");
+    EXPECT_EQUAL("", v000s.toAbbreviatedString());
+    EXPECT_EQUAL("0.0.0", v000s.toString());
+
+    Version v100(1, 0, 0, "");
+    EXPECT_EQUAL("1", v100.toAbbreviatedString());
+    EXPECT_EQUAL("1.0.0", v100.toString());
+    Version v100s("1");
+    EXPECT_EQUAL("1", v100s.toAbbreviatedString());
+    EXPECT_EQUAL("1.0.0", v100s.toString());
+
+    Version v110(1, 1, 0, "");
+    EXPECT_EQUAL("1.1", v110.toAbbreviatedString());
+    EXPECT_EQUAL("1.1.0", v110.toString());
+    Version v110s("1.1");
+    EXPECT_EQUAL("1.1", v110s.toAbbreviatedString());
+    EXPECT_EQUAL("1.1.0", v110.toString());
+
+    Version v111(1, 1, 1, "");
+    EXPECT_EQUAL("1.1.1", v111.toAbbreviatedString());
+    EXPECT_EQUAL("1.1.1", v111.toString());
+    Version v111s("1.1.1");
+    EXPECT_EQUAL("1.1.1", v111s.toAbbreviatedString());
+    EXPECT_EQUAL("1.1.1", v111s.toString());
```

Comment:
It may border on paranoia to do so, but it could be useful to specify distinct numbers for the major/minor/micro parts of the version string (e.g. `1.2.3`) to catch any unintended errors caused by e.g. reusing one parsed component across multiple fields.